### PR TITLE
minor fixes

### LINKED ISSUE #880: Bug: Mismatch in Studio and Engine handling of unresolved class mapping ID
### Similar issues

- [X] I have searched and found no existing similar issues

### How are you using Studio?

Legend Studio

### Current and expected behavior

A number of Pure users have mappings which contain class mapping IDs pointing at another mapping's class mappings. In Pure/Engine, we let this pass compilation phase. In Studio, we disallow this. This makes it hard for users to migrate to Studio. We should think of a strategy to make things loadable in Studio, but disallow users to make changes if they have this kind of error in their graph.

### Steps to reproduce

Use the following grammar. Try compilation in text mode, Engine will pass while Studio graph builder will scream.

### Model data

```text
###Relational
Database model::Test
(
  Table FirmTable
  (
    id INTEGER PRIMARY KEY,
    Legal_name VARCHAR(200)
  )
  Table PersonTable
  (
    id INTEGER PRIMARY KEY,
    firm_id INTEGER,
    firstName VARCHAR(200),
    lastName VARCHAR(200)
  )

  Join FirmPerson(PersonTable.firm_id = FirmTable.id)
)


###Pure
Profile model::MyExtension
{
  stereotypes: [important];
  tags: [doc];
}

Enum model::IncType
{
  Corp,
  LLC
}

Class model::LegalEntity
{
  legalName: String[1];
}

Class model::Person
{
  firstName: String[1];
  lastName: String[1];
}

Class <<model::MyExtension.important>> {model::MyExtension.doc = 'This is a model of a firm'} model::Firm extends model::LegalEntity
[
  validName: $this.legalName->startsWith('')
]
{
  employees: model::Person[1..*];
  incType: model::IncType[1];
  employeeSize() {$this.employees->count()}: Number[1];
}


###Mapping
Mapping model::NewMapping
(
  *model::Firm: Relational
  {
    ~primaryKey
    (
      [model::Test]FirmTable.id
    )
    ~mainTable [model::Test]FirmTable
    legalName: concat([model::Test]FirmTable.Legal_name, '_LTD'),
    employees[idNotFound]: [model::Test]@FirmPerson
  }
  *model::Person: Relational
  {
    ~primaryKey
    (
      [model::Test]PersonTable.id
    )
    ~mainTable [model::Test]PersonTable
    firstName: [model::Test]PersonTable.firstName,
    lastName: [model::Test]PersonTable.lastName
  }
)
```

### Environment

```text
System:
    OS: macOS 11.6.4
  Binaries:
    Node: 16.8.0 - /private/var/folders/95/y32364512q77c1dvhkpnr_n80000gn/T/xfs-e9ca549e/node
    Yarn: 3.2.0 - /private/var/folders/95/y32364512q77c1dvhkpnr_n80000gn/T/xfs-e9ca549e/yarn
    npm: 7.21.0 - ~/.nvm/versions/node/v16.8.0/bin/npm
  Browsers:
    Chrome: 98.0.4758.102
    Firefox: 96.0.3
```

### Possible solution and workaround

Discussed with @pierredebelen and @MauricioUyaguari. We will not hard-fail in Studio, the general strategy is detailed in https://github.com/finos/legend-studio/issues/941

### Contribution

- [ ] I would like to work on the fix for this issue

### PR DESCRIPTION
**Pull request by akphi**
## Summary

- [x] update `CONTRIBUTING.md`
- [x] temporarily disable failing graph building when detected unresolved class mapping IDs - partially resolves #880 
- [x] bump dependencies
- [x] revert to fs-extra@10.0.0 to avoid publishing issue and re-trigger publish
- [x] use 'fixed' instead of 'linked' mode for changeset - See https://github.com/changesets/changesets/pull/690

## How did you test this change?

- [ ] Test(s) added
- [ ] Manual testing (please provide screenshots/recordings)
- [x] No testing (please provide an explanation)

### CODE REVIEW

File: fixtures/legend-docker-setup/end-to-end-setup/end-to-end-setup-docker-compose.yml [hunk 1]
```diff
@@ -0,0 +1,22 @@
...
+    # NOTE: the Java CLI wildcard must be wrapped by quotes
+    # See https://stackoverflow.com/questions/14722657/java-classpath-wildcard-behaviour/14722763
+    command: >
+      java
+      -cp /app/bin/"*"
+      org.finos.legend.engine.server.Server
+      server /config/engine-config.json
+
+networks:
+  legend: {}
```
Comment (MauricioUyaguari) [inline fixtures/legend-docker-setup/end-to-end-setup/end-to-end-setup-docker-compose.yml:22]: Is there a changset on this? what do your docker changes do in this change ?

Comment (akphi) [reply fixtures/legend-docker-setup/end-to-end-setup/end-to-end-setup-docker-compose.yml:22]: no, this is the placeholder only

File: packages/legend-query/src/components/QueryBuilderExplorerPanel.tsx [hunk 2]
```diff
@@ -244,14 +244,14 @@ const QueryBuilderExplorerPropertyDragLayer = observer(
...
+    HTMLDivElement,
+    {
       queryBuilderState: QueryBuilderState;
       openNode: () => void;
       node: QueryBuilderExplorerTreeNodeData;
-    },
-    ref: React.Ref<HTMLDivElement>,
-  ) => {
+    }
+  >(function QueryBuilderExplorerContextMenu(props, ref) {
```
Comment (MauricioUyaguari) [inline packages/legend-query/src/components/QueryBuilderExplorerPanel.tsx:254]: Can we mentioned why we are doing this? Due to an upgrade in dependency?

File: packages/legend-graph/src/__tests__/buildGraph/GraphBuildFailure.test.ts [hunk 3]
```diff
@@ -204,7 +207,7 @@ test.skip(unitTest('Duplicate class properties'), async () => {
   );
 });
 
-/* @MARKER: Temporary until we resolve https://github.com/finos/legend-studio/issues/660 */
+/* @MARKER: RELAXED GRAPH CHECK - See https://github.com/finos/legend-studio/issues/660 */
 test.skip(unitTest('Duplicate aoociation properties'), async () => {
```
Comment (MauricioUyaguari) [inline packages/legend-graph/src/__tests__/buildGraph/GraphBuildFailure.test.ts:211]: could we fix this typo `aoociation`

File: packages/legend-query/src/components/QueryBuilderExplorerPanel.tsx (hunk 2 above)
Comment (akphi) [reply packages/legend-query/src/components/QueryBuilderExplorerPanel.tsx:254]: yes, this is warned by `mobx-react-lite`

```
[mobx-react-lite] `observer(fn, { forwardRef: true })` is depreacted, use `observer(React.forwardRef(fn))`
```

**Review by MauricioUyaguari — DISMISSED** [review]

**Review by MauricioUyaguari — APPROVED** [review]

**Pull request merged**